src/wymeditor/editor/dialogs.js

```diff
@@ -0,0 +1,568 @@
+"use strict";
+/**
+    editor.dialog
+    =============
+
+    Open a dialog box
+*/
+WYMeditor.editor.prototype.dialog = function (
+    dialog,
+    pDialogWindowFeatures,
+    pBodyHtml
+) {
+    var wym = this,
+        i,
+        DIALOGS = WYMeditor.DIALOGS,
+        dialogObject,
+        dialogWindowFeatures,
+        wDialog,
+        htmlStrReplacements,
+        dialogHtml,
+        doc,
+        selectedContainer,
+        options = wym._options;
+
```

Comment:
Some changes below, to accept a `dialog` object while maintaining support for the old way.